ui/src/app/components/apps-panel.tsx

```diff
@@ -1,3 +1,139 @@
 import * as React from 'react';
+import { useState } from 'react';
 
-export const AppsPanel = () => <div>TODO</div>;
+import { BillingStatus, Workspace } from 'generated/fetch';
+
+import { Clickable, CloseButton } from 'app/components/buttons';
+import { FlexColumn, FlexRow } from 'app/components/flex';
+import { DisabledPanel } from 'app/components/runtime-configuration-panel/disabled-panel';
+import colors from 'app/styles/colors';
+import { reactStyles } from 'app/utils';
+import { isVisible } from 'app/utils/runtime-utils';
+import { runtimeStore, useStore } from 'app/utils/stores';
+
+import { AppLogo } from './apps-panel/app-logo';
+import { ExpandedApp } from './apps-panel/expanded-app';
+import { UIAppType } from './apps-panel/utils';
+
+const styles = reactStyles({
+  header: {
+    color: colors.primary,
+    height: 19,
+    fontFamily: 'Montserrat',
+    fontSize: 16,
+    fontWeight: 600,
+    letterSpacing: 0,
+    lineHeight: '19px',
+    margin: '1em',
+  },
+  availableApp: {
+    background: colors.white,
+    marginLeft: '1em',
+    marginBottom: '1em',
+    justifyContent: 'center',
+  },
+  closeButton: { marginLeft: 'auto', alignSelf: 'center' },
+});
+
+const AvailableApp = (props: { appType: UIAppType; onClick: Function }) => {
+  const { appType, onClick } = props;
+  return (
+    <Clickable {...{ onClick }}>
+      <FlexRow style={styles.availableApp}>
+        <AppLogo {...{ appType }} />
+      </FlexRow>
+    </Clickable>
+  );
+};
+
+export const AppsPanel = (props: {
+  workspace: Workspace;
+  onClose: Function;
+  onClickRuntimeConf: Function;
+  onClickDeleteRuntime: Function;
+}) => {
+  const { onClose } = props;
+  const { runtime } = useStore(runtimeStore);
+
+  // in display order
+  const appsToDisplay = [UIAppType.JUPYTER, UIAppType.RSTUDIO];
+
+  const appStates = [
+    { appType: UIAppType.JUPYTER, expand: isVisible(runtime?.status) },
+    // RStudio is not implemented yet, so we don't expand it
+    { appType: UIAppType.RSTUDIO, expand: false },
+  ];
+
+  // which app(s) have the user explicitly expanded by clicking?
+  const [userExpandedApps, setUserExpandedApps] = useState([]);
+  const addToExpandedApps = (appType: UIAppType) =>
+    setUserExpandedApps([...userExpandedApps, appType]);
+
+  const showExpanded = (appType: UIAppType): boolean =>
+    userExpandedApps.includes(appType) ||
```

Comment:
Should we reverse the check i.e appStates.find((s) => s.appType === appType)?.expand is checked first and then includes? The reason i say that is because in 2 places we are explicitly checking userExpandedApps.includes(appType).